src/test/java/com/segment/analytics/android/integrations/google/analytics/GoogleAnalyticsTest.java

```diff
@@ -204,6 +207,47 @@ static AnalyticsContext contextWithCampaign(Campaign campaign) throws Exception
         .build());
   }
 
+  @Test public void trackWithCustomDimensionsAndProducts() {
```

Comment:
Nit pick but I would add more e-commerce events formatted to our spec. The product viewed event below seems to trigger products at the top level and within products, is this expected?

Reply:
Yes, this is to keep backward compatibility with the current behaviour.

> To preserve backward compatibility, customDimensions on the product hit are set twice (once for the product map, and once for the event map).

Reply:
Would it still make sense to add another test case?

I would test:

1. case where ecommerce event only has product info in product
2. case where ecommerce event only has product info as top level properties
3. case where non-standard (not spec'd ecommerce) event is sent but with properites

Reply:
👍 that's a good idea!

Reply:
done!